diff -r e2b2b5d56e1a -r b08c9ee9d69b browser/app/nsBrowserApp.cpp
--- a/browser/app/nsBrowserApp.cpp	Mon Mar 16 12:01:19 2020 -0400
+++ b/browser/app/nsBrowserApp.cpp	Mon Mar 16 12:14:25 2020 -0400
@@ -164,7 +164,6 @@
       Output("Couldn't set %s.\n", appEnv);
       return 255;
     }
-    //
     argv[2] = argv[0];
     argv += 2;
     argc -= 2;
@@ -186,12 +185,10 @@
   if (appDataFile && *appDataFile) {
     config.appData = nullptr;
     config.appDataPath = appDataFile;
-    
   } else {
     // no -app flag so we use the compiled-in app data
     config.appData = &sAppData;
     config.appDataPath = kDesktopFolder;
-    
   }
 
 #if defined(XP_WIN) && defined(MOZ_SANDBOX)
@@ -210,6 +207,7 @@
 #ifdef LIBFUZZER
   if (getenv("FUZZER"))
     gBootstrap->XRE_LibFuzzerSetDriver(fuzzer::FuzzerDriver);
+
 #endif 
 
   // Note: keep in sync with LauncherProcessWin.
@@ -217,7 +215,6 @@
   EnsureCommandlineSafe(argc, argv, acceptableParams);
 
   return gBootstrap->XRE_main(argc, argv, config);
-
 }
 
 static nsresult InitXPCOMGlue(LibLoadingStrategy aLibLoadingStrategy) {
